### PR TITLE
Create TripTest and Implement getMaxDay()

### PR DESCRIPTION
**Pull request by ivanderjmw**
Some other minor features like fix index numbering, error handling for new trip, fix IO redirection and exceptions.

### CODE REVIEW

File: src/main/java/seedu/trippie/data/Trip.java [hunk 1]
```diff
@@ -9,20 +12,36 @@
...
     }
 
-    public Trip(int index, String name, Date startDate) {
+    public Trip(int index, String name, Date startDate) throws TrippieException {
+
+        if (index < 0) {
+            throw new TrippieException("Index out of bounds");
+        }
+
+        if (name.length() == 0) {
```
Comment (felixhalim) [inline src/main/java/seedu/trippie/data/Trip.java:28]: An alternative way is to use `.isEmpty()` method

File: src/test/java/seedu/trippie/data/TripTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    @Test
+    void getMaxDay() {
+    }
+
+    @Test
+    void testToString() {
+    }
+}
```
Comment (felixhalim) [inline src/test/java/seedu/trippie/data/TripTest.java:85]: Please add `newline` here

File: src/test/java/seedu/trippie/data/TripTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        Trip trip = new Trip();
+        PlaceList placeList = new PlaceList();
+
+        assertNotEquals(trip.getPlaceListObject(), placeList);
+        trip.setPlaceList(placeList);
+        assertEquals(trip.getPlaceListObject(), placeList);
+    }
+
+    @Test
+    void getMaxDay() {
```
Comment (felixhalim) [inline src/test/java/seedu/trippie/data/TripTest.java:79]: Perhaps deleting this method as we aren't going to perform any test?

**Review by felixhalim — COMMENTED** [review]
Great job on implementing testing on `Trip.java`. Please review some of my comments.

**Comment by ivanderjmw** [comment]
Thank you felix for referencing the issue.

**Review by ShawnTanzc — APPROVED** [review]
LGTM

**Pull request merged**